### DISCUSSION TITLE
How can I get the characters corresponding to the current chunk in a streaming speech synthesis?

### DISCUSSION THREAD

**Discussion by jiayueshiliuqi** [coqui-ai/TTS#4055] [General Q&A]
I currently have the following streaming code:

@app.get("/synthesize")
async def synthesize(text: str):
    chunks = model.inference_stream(
        text,
        "zh",
        gpt_cond_latent,
        speaker_embedding
    )

    def audio_stream():
        print('start stream')
        for i, chunk in enumerate(chunks):
            chunk_bytes = chunk.cpu().numpy().tobytes()

            buf = io.BytesIO()
            buf.write(chunk_bytes)

            buf.seek(0)
            yield buf.read()
        print('Done')

    return StreamingResponse(audio_stream(), media_type="audio/wav")

I want to get the characters corresponding to each chunk. For example, if the input text is "在很久很久以前", the chunks might be "在很久", "很久", and "以前". Is it possible to get and return the corresponding characters for each chunk?

**Comment by bharathraj-v** [comment]
@jiayueshiliuqi, did you figure this out? If so can you please share how you've done it - I'm also looking for the same functionality
